### PR TITLE
Task nxp 29691 jenkins pipeline

### PR DESCRIPTION
**Pull request by ataillefer**

### CODE REVIEW

**Review by akervern — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+}
+
+pipeline {
+  agent {
+    label 'jenkins-nuxeo-package-11'
+  }
+  triggers {
+    upstream(
+      threshold: hudson.model.Result.SUCCESS,
+      upstreamProjects: "/nuxeo/nuxeo/master",
```
Comment (atchertchian) [inline Jenkinsfile:83]: Could use `${BRANCH_NAME}` to cover potential maintenance branches use case (but that assumes branches will be named the same, and that's a detail, there will be other changes needed anyway to cover this)

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  step([
+    $class: 'GitHubCommitStatusSetter',
+    reposSource: [$class: 'ManuallyEnteredRepositorySource', url: 'https://github.com/nuxeo/nuxeo-customer-project-sample'],
+    contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
+    statusResultSource: [$class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]],
+  ])
+}
+
+void getNuxeoImageVersion() {
+  return sh(returnStdout: true, script: 'mvn org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression=nuxeo.platform.version -q -DforceStdout').trim();
```
Comment (atchertchian) [inline Jenkinsfile:38]: So you confirm that this works for you (only when using 3.1.0)? :thinking:

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+          Registry: ${DOCKER_REGISTRY}
+          """
+          withEnv(["NUXEO_IMAGE_VERSION=${getNuxeoImageVersion()}"]) {
+            withCredentials([string(credentialsId: 'instance-clid', variable: 'INSTANCE_CLID')]) {
+              script {
+                // build and push Docker image to the Jenkins X internal Docker registry
+                def dockerPath = 'docker/nuxeo-customer-project-sample-docker'
+                sh "envsubst < ${dockerPath}/skaffold.yaml > ${dockerPath}/skaffold.yaml~gen"
+                sh """#!/bin/bash +x
+                  CLID=\$(echo -e "${INSTANCE_CLID}" | sed ':a;N;\$!ba;s/\\n/--/g') skaffold build -f ${dockerPath}/skaffold.yaml~gen
```
Comment (atchertchian) [inline Jenkinsfile:186]: Aren't you leaking the clid here?

**Review by atchertchian — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (ataillefer) [reply Jenkinsfile:38]: Working, yes

File: Jenkinsfile (hunk 3 above)
Comment (ataillefer) [reply Jenkinsfile:186]: Nope thanks to `#!/bin/bash +x`.

File: Jenkinsfile (hunk 1 above)
Comment (ataillefer) [reply Jenkinsfile:83]: Yep, for now I think we can stick to the Cloud release only.

**Pull request merged**